src/ServiceControl/CompositeViews/AuditCounts/GetAuditCountsForEndpointApi.cs

```diff
@@ -19,13 +19,20 @@ public record AuditCountsForEndpointContext(PagingInfo PagingInfo, string Endpoi
     public class GetAuditCountsForEndpointApi(
         IErrorMessageDataStore dataStore,
         Settings settings,
-        IHttpClientFactory httpClientFactory,
-        IHttpContextAccessor httpContextAccessor)
+        IHttpClientFactory httpClientFactory
+    )
         : ScatterGatherApi<IErrorMessageDataStore, AuditCountsForEndpointContext, IList<AuditCount>>(dataStore, settings,
-            httpClientFactory, httpContextAccessor)
+            httpClientFactory)
     {
         static readonly IList<AuditCount> Empty = new List<AuditCount>(0).AsReadOnly();
 
+        protected override string Url(AuditCountsForEndpointContext input) =>
```

Comment:
I can see why you are inlining the URI templates here from the perspective of your work. There is a larger implication though that comes with this change. One of the primary concept of the forwarding and the scatter gather approaches in SC is to take a request _as is_  and forward that to the downstream instance. This is also the perspective under which we have evaluated alternative solutions like YARP for forwarding and also scatter gather (which we decided against using it there). Still the underlying premise is the same. You want to take a request and forward it down. By in lining the template creation Api class based on the data of the context, the relationship between the destination URI and the context data is now hard-coded. This means we are loosening the benefit of being able to forward ANY type of contextual information that previously was on the HTTP context. This also has consequences for any sort of header information we might want to flow across instances at a later point in time. 

I do believe this should be further discussed.

Reply:
It could cause issues, given the current expectations.
I am trying to figure out if there is a way to test it, but it isn't easy.
On the other hand, we can't guarantee that the destination will have a match when we forward the request to the destination, so there is no really nice way to validate it.
Also, in theory a customer could be using different version of all instances, eg monitoring v1, SC primary v3, audit v2, I don't think we do any kind of enforcement to update everything together, so there could be mismatches.

So again, back to our requirement.
We need to use scatter-gather from non-HTTP context, and it feels weird for the caller of the API needs to know the URL path and querystrings.
So yes, maybe this proposal needs to be more discussed, I assume you referring to RFCing it?

Reply:
So @danielmarbach, after talking with @jpalac we decided to remove the URL refactor and instead just do the initial commit, which does not change the URL at all.
As far as I understand, there were no issues with it, so I am going to push ahead and merge that one so we can proceed in https://particularsoftware.slack.com/archives/C06CUKQ3668